src/endpoints/proposals.ts

```diff
@@ -165,5 +199,47 @@ export const proposals = (
         ));
     }
 
+    // Save the proposal to the table
+    const proposalId = ccf.crypto.digest("SHA-256", request.body.arrayBuffer());
+    proposalsPolicyMap.set(
+        proposalId,
+        request.body.arrayBuffer()
+    );
+    createdTimeToProposalIdMap.set(currentProposalCreatedAt, proposalId);
+
+    // Keep the last N proposals
+    const sortedCreatedTimes = Array.from(createdTimeToProposalIdMap.keys()).sort((a, b) => a - b);
+    for (let i = 0; i < sortedCreatedTimes.length - proposalsToKeep; i++) {
+        proposalsPolicyMap.delete(createdTimeToProposalIdMap.get(sortedCreatedTimes[i])!);
+    }
+
     return ServiceResult.Succeeded<IProposalResult[]>(proposalResults, logContext);
+}
+
+export const getProposals = (
+    request: ccfapp.Request<IProposalsRequest>,
+  ): ServiceResult<string[]> => {
+    const logContext = new LogContext().appendScope("proposals");
+    let proposals: string[] = [];
+
+    // Sort the proposals by created time
```

Comment:
Is it because it's necessary? Asking just to understand

Reply:
It isn't strictly necessary the client could do the sorting themselves, but I think it's more convenient to do it on the server side

Reply:
Got it thanks. You might want to document it somewhere "public" (vs implementation.) like app.json if you want to let clients take advantage of it (otherwise it's happened to be sorted from client point of view). But I think it's up to you.

Reply:
Yeah if app.json had a "description" field that would be the perfect place, for now I've added it to the big comment at the top of proposals.ts

https://github.com/microsoft/azure-privacy-sandbox-kms/pull/457/commits/f348d89776afebb56ae53ad495d758203204e5c8